mongorestore/mongorestore_test.go

```diff
@@ -250,6 +251,110 @@ func TestMongorestore(t *testing.T) {
+			So(db.WriteConcern(), ShouldResemble, writeconcern.New(writeconcern.WMajority()))
+		})
+
+		longCollection := db.Collection(specialCharactersCollectionName)
+		longCollection.Drop(nil)
+
+		Convey("and --nsInclude a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSInclude = make([]string, 1)
+			restore.NSOptions.NSInclude[0] = "db1." + specialCharactersCollectionName
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 1)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := longCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 1)
+
+		})
+
+		Convey("and --nsExclude a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSExclude = make([]string, 1)
+			restore.NSOptions.NSExclude[0] = "db1." + specialCharactersCollectionName
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 0)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := longCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
+
+		})
+
+		Convey("and --nsTo a collection name without special characters "+
+			"--nsFrom a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSFrom = make([]string, 1)
+			restore.NSOptions.NSFrom[0] = "db1." + specialCharactersCollectionName
+			restore.NSOptions.NSTo = make([]string, 1)
+			restore.NSOptions.NSTo[0] = "db1.aCollectionNameWithoutSpecialCharacters"
+
+			shortCollection := db.Collection("aCollectionNameWithoutSpecialCharacters")
+			shortCollection.Drop(nil)
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 1)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := shortCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 1)
+
+		})
+
+		Convey("and --nsTo a collection name with special characters "+
+			"--nsFrom a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSFrom = make([]string, 1)
+			restore.NSOptions.NSFrom[0] = "db1." + specialCharactersCollectionName
+			restore.NSOptions.NSTo = make([]string, 1)
+			restore.NSOptions.NSTo[0] = "db1.aCollectionNameWithSpećiálCharacters"
+
+			shortCollection := db.Collection("aCollectionNameWithSpećiálCharacters")
+			shortCollection.Drop(nil)
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 1)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := shortCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 1)
+
```

Comment:
[super-nit] I noticed a lot of blank lines at the end of the convey blocks; can you remove these?

Reply:
Will do!